FindBaselibs.cmake

```diff
@@ -1,97 +1,131 @@
-    execute_process (COMMAND ${CMAKE_C_COMPILER} --print-file-name=libgcc.a OUTPUT_VARIABLE libgcc OUTPUT_STRIP_TRAILING_WHITESPACE)
-  endif()
-else ()
-  execute_process (COMMAND ${CMAKE_CXX_COMPILER} --print-file-name=libstdc++.so OUTPUT_VARIABLE stdcxx OUTPUT_STRIP_TRAILING_WHITESPACE)
-endif ()
-
-
-# We must statically link ESMF on Apple due mainly to an issue with how Baselibs is built.
-# Namely, the esmf dylib libraries end up with the full *build* path on Darwin (which is in 
-# src/esmf/lib/libO...) But we copy the dylib to $BASEDIR/lib. Thus, DYLD_LIBRARY_PATH gets
-# hosed. yay.
-if (APPLE)
-   set (ESMF_LIBRARY ${BASEDIR}/lib/libesmf.a)
-else ()
-   set (ESMF_LIBRARY esmf_fullylinked)
-endif ()
-
-#find_package (NetCDF REQUIRED COMPONENTS Fortran)
-#set (INC_NETCDF ${NETCDF_INCLUDE_DIRS})
-
-set (NETCDF_LIBRARIES ${NETCDF_LIBRARIES_OLD})
-set (ESMF_LIBRARIES ${ESMF_LIBRARY} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES} ${MPI_CXX_LIBRARIES} ${stdcxx} ${libgcc})
-
-# BASEDIR.rc file does not have the arch
-string(REPLACE "/${CMAKE_SYSTEM_NAME}" "" BASEDIR_WITHOUT_ARCH ${BASEDIR})
-set(BASEDIR_WITHOUT_ARCH ${BASEDIR_WITHOUT_ARCH} CACHE STRING "BASEDIR without arch")
-mark_as_advanced(BASEDIR_WITHOUT_ARCH)
-
-# Set the site variable
-include(DetermineSite)
+if (Baselibs_FOUND)
+
+  link_directories (${BASEDIR}/lib)
+
+  # Add path to GFE packages
+  list (APPEND CMAKE_PREFIX_PATH ${BASEDIR})
+
+  #------------------------------------------------------------------
+  # netcdf
+  # The following command provides the list of libraries that netcdf
+  # uses.  Unfortunately it also includes the library path and "-l"
+  # prefixes, which CMake handles in a different manner. So we need so
+  # strip off that item from the list
+  execute_process (
+    COMMAND ${BASEDIR}/bin/nf-config --flibs
+    OUTPUT_VARIABLE LIB_NETCDF
+    )
+
+  string(REGEX MATCHALL " -l[^ ]*" _full_libs "${LIB_NETCDF}")
+  set (NETCDF_LIBRARIES_OLD)
+  foreach (lib ${_full_libs})
+    string (REPLACE "-l" "" _tmp ${lib})
+    string (STRIP ${_tmp} _tmp)
+    list (APPEND NETCDF_LIBRARIES_OLD ${_tmp})
+  endforeach()
+
+  list (REVERSE NETCDF_LIBRARIES_OLD)
+  list (REMOVE_DUPLICATES NETCDF_LIBRARIES_OLD)
+  list (REVERSE NETCDF_LIBRARIES_OLD)
+
+  add_definitions(-DHAS_NETCDF4)
+  add_definitions(-DHAS_NETCDF3)
+  add_definitions(-DH5_HAVE_PARALLEL)
+  add_definitions(-DNETCDF_NEED_NF_MPIIO)
+  add_definitions(-DHAS_NETCDF3)
+  #------------------------------------------------------------------
+
+  set (INC_HDF5 ${BASEDIR}/include/hdf5)
+  set (INC_NETCDF ${BASEDIR}/include/netcdf)
+  set (INC_HDF ${BASEDIR}/include/hdf)
+  set (INC_ESMF ${BASEDIR}/include/esmf)
+
+  find_package(GFTL REQUIRED)
+  find_package(GFTL_SHARED REQUIRED)
+  find_package(FARGPARSE QUIET)
+  find_package(YAFYAML REQUIRED)
+  find_package(PFLOGGER REQUIRED)
+
+  find_package(FLAP REQUIRED)
```

Comment:
Hmm.  I thought we had an option where FLAP was not mandatory  (@mathomp4)

Reply:
Ugh. Yeah. Looking around, It's sort of half-baked. So MAPL has:
```cmake
option(BUILD_WITH_FLAP "Use FLAP for command line processing" ON)
```
But there seems to be no control around for the `find_package(FLAP)`.

What we should do is remove `find_package(FLAP)` from here and do something like:
```cmake
option(BUILD_WITH_FLAP "Use FLAP for command line processing" ON)
if(BUILD_WITH_FLAP)
  find_package(FLAP REQUIRED)
endif()
```
or the like. MAPL needs FLAP, so it should ask for it. I suppose many of these other `find_package()` should migrate as well. Technically it's MAPL that needs them, not GEOS.

Reply:
Though, I suppose if we use this setup, it should also have a `Baselibs_FOUND` test.

Reply:
Do we agree on removing `find_package(FLAP REQUIRED)` from this `CMakeLists.txt` and add a test to MAPL as @mathomp4 suggests? I've verified that this change would work for us.

Since FLAP isn't mandatory, should we set `BUILD_WITH_FLAP` to `OFF` by default?